javatests/arcs/core/storage/handle/SingletonIntegrationTest.kt

```diff
@@ -0,0 +1,170 @@
+        StorageProxy<EntitySingletonData, EntitySingletonOp, EntitySingletonView>
+    private lateinit var singletonA: SingletonImpl<RawEntity>
+    private lateinit var singletonB: SingletonImpl<RawEntity>
+
+    @Before
+    fun setUp() = runBlocking {
+        RamDiskDriverProvider()
+
+        store = Store(STORE_OPTIONS)
+        storageProxy = StorageProxy(store.activate(), CrdtSingleton<RawEntity>())
+
+        singletonA = SingletonImpl("singletonA", storageProxy)
+        storageProxy.registerHandle(singletonA)
+        singletonB = SingletonImpl("singletonB", storageProxy)
+        storageProxy.registerHandle(singletonB)
+        Unit
+    }
+
+    @After
+    fun tearDown() {
+        RamDisk.clear()
+    }
+
+    @Test
+    fun initialState() = runBlockingTest {
+        assertThat(singletonA.value()).isNull()
+        assertThat(singletonB.value()).isNull()
+    }
+
+    @Test
+    fun settingOnA_showsUpInB() = runBlockingTest {
+        val person = Person("Lou", 95, true)
+
+        singletonA.set(person.toRawEntity())
+        assertThat(singletonA.value()).isEqualTo(person.toRawEntity())
+        assertThat(singletonB.value()).isEqualTo(person.toRawEntity())
+    }
+
+    @Test
+    fun reSettingOnB_showsUpInA() = runBlockingTest {
+        val lou = Person("Lou", 95, true)
+        val jan = Person("Jan", 28, true, emptySet())
+
+        singletonA.set(lou.toRawEntity())
+        assertThat(singletonA.value()).isEqualTo(lou.toRawEntity())
+        assertThat(singletonB.value()).isEqualTo(lou.toRawEntity())
+
+        singletonB.set(jan.toRawEntity())
+        assertThat(singletonA.value()).isEqualTo(jan.toRawEntity())
+        assertThat(singletonB.value()).isEqualTo(jan.toRawEntity())
+    }
+
+    @Test
+    fun clearingOnA_clearsB() = runBlockingTest {
+        val person = Person("Susan", 48, true)
+
+        singletonA.set(person.toRawEntity())
+        assertThat(singletonA.value()).isEqualTo(person.toRawEntity())
+        assertThat(singletonB.value()).isEqualTo(person.toRawEntity())
+
+        singletonB.clear()
+        assertThat(singletonA.value()).isNull()
+        assertThat(singletonB.value()).isNull()
+    }
+
+    private data class Person(
+        val name: String,
+        val age: Int,
+        val isCool: Boolean,
+        val pets: Set<String> = setOf("Fido")
+    ) {
+        fun toRawEntity(): RawEntity = RawEntity(
+            hashCode().toString(),
+            singletons = mapOf(
+                "name" to name.toReferencable(),
+                "age" to age.toReferencable(),
+                "is_cool" to isCool.toReferencable()
+            ),
+            collections = mapOf(
+                "pets" to pets.map { it.toReferencable() }.toSet()
+            )
+        )
+    }
+
+    companion object {
+        private val STORAGE_KEY = ReferenceModeStorageKey(
+            backingKey = RamDiskStorageKey("entities"),
+            storageKey = RamDiskStorageKey("entity_collection")
```

Comment:
Consider using a different storage key here, AFAIK we can't guarantee that these two test classes won't be run in parallel on the same host. To the extent this is an example for clients, we can also encourage good naming practices. :)